pkg/client/invokeandexecute.go

```diff
@@ -29,7 +29,12 @@ func invokeAndExecute[Caveats, Out any](
 	successType schema.Type,
 	options ...delegation.Option,
 ) (result.Result[Out, failure.IPLDBuilderFailure], fx.Effects, error) {
-	inv, err := capParser.Invoke(c.Issuer(), c.Connection().ID(), with, caveats, options...)
+	pfs := make([]delegation.Proof, 0, len(c.Proofs()))
+	for _, del := range c.Proofs() {
+		pfs = append(pfs, delegation.FromDelegation(del))
+	}
+
+	inv, err := capParser.Invoke(c.Issuer(), c.Connection().ID(), with, caveats, append(options, delegation.WithProof(pfs...))...)
```

Comment:
very little nitpick:
```suggestion
	options = append(options, delegation.WithProof(pfs...))
	inv, err := capParser.Invoke(c.Issuer(), c.Connection().ID(), with, caveats, options...)
```

Reply:
Ah! 👍🏻